### PR TITLE
iter: tiny cleanup

### PR DESCRIPTION
**Pull request by BusyJay**
@siddontang @zhangjinpeng1987 @disksing PTAL

### CODE REVIEW

File: src/raftstore/coprocessor/region_snapshot.rs [hunk 1]
```diff
@@ -156,16 +149,19 @@ fn adjust_upper_bound(upper_bound: Option<&[u8]>) -> Option<&[u8]> {
     }
 }
 
+fn adjust_iter_opt(iter_opt: IterOption, region: &Region) -> IterOption {
```
Comment (ngaut) [inline src/raftstore/coprocessor/region_snapshot.rs]: s/adjust_iter_opt/set_upper_bound

File: src/raftstore/store/engine.rs [hunk 2]
```diff
@@ -151,33 +151,43 @@ pub trait Peekable {
...
+    pub fn new(upper_bound: Option<Vec<u8>>, fill_cache: bool) -> IterOption {
         IterOption {
             upper_bound: upper_bound,
             fill_cache: fill_cache,
-            seek_mode: seek_mode,
+            seek_mode: SeekMode::TotalOrder,
         }
     }
 
+    pub fn use_prefix_seek(mut self) -> IterOption {
```
Comment (siddontang) [inline src/raftstore/store/engine.rs]: do we need another `use_total_order_seek` function if we want to change the seek mode dynamically?

Comment (BusyJay) [inline src/raftstore/store/engine.rs]: Yes. But generally we won't use it in this way.

File: src/raftstore/coprocessor/region_snapshot.rs [hunk 3]
```diff
@@ -148,24 +141,21 @@ pub struct RegionIterator<'a> {
...
 
-fn adjust_upper_bound(upper_bound: Option<&[u8]>) -> Option<&[u8]> {
-    if let Some(k) = upper_bound {
-        if k.is_empty() { None } else { Some(k) }
-    } else {
-        None
-    }
+fn set_upper_bound(iter_opt: IterOption, region: &Region) -> IterOption {
+    let upper_bound = match iter_opt.upper_bound() {
+        Some(k) if !k.is_empty() => keys::data_key(k),
```
Comment (siddontang) [inline src/raftstore/coprocessor/region_snapshot.rs]: why need to set upper bound below if the IterOption already has the upper bound?

Comment (BusyJay) [inline src/raftstore/coprocessor/region_snapshot.rs]: Because the key has to be encoded with the magic prefix 'z'.

**Comment by BusyJay** [comment]
@zhangjinpeng1987 @disksing PTAL

**Comment by siddontang** [comment]
Ping @zhangjinpeng1987

**Comment by ngaut** [comment]
LGTM

**Pull request merged**